internal/server/server.go

```diff
@@ -81,6 +81,7 @@ func (s *Server) Run() error {
 		middleware.Logger,
 		middleware.Recoverer,
 		oapimiddleware.OapiRequestValidatorWithOptions(swagger, &oapiOpts),
+		// auth.ParseHeaders,
```

Comment:
We need to gate the activation of this with a config option for users to explicitly "opt in". This must only be activated when deployed behind an API gateway that removes headers from all incoming requests and sets the headers new itself.

Reply:
@akoserwal, that would make sense...